File: thesis-core/src/main/java/it/nicolalopatriello/thesis/core/dto/repository/RepositoryDetails.java
package it.nicolalopatriello.thesis.core.dto.repository;

import it.nicolalopatriello.thesis.common.dto.Metric;
import it.nicolalopatriello.thesis.core.dto.Vulnerability;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class RepositoryDetails {
    private Long id;
    private String url;
    private String branch;
    private String lastCommitSha;
    private String summary;
    private Long runnerFinishedAt;
    private Long minutesWatchersInterval;
    private List<DependencyWithVulnerabilities> dependencies;
    private List<Metric> metrics;

    @Getter
    @Setter
    public static class DependencyWithVulnerabilities {
        private Long id;
        private String name;
        private String version;
        private String programmingLanguage;
        private List<Vulnerability> vulnerabilities;
    }
}

File: thesis-ui/src/app/@core/models/repository-details.ts
export interface RepositoryDetails {
  id: number;
  url: string;
  branch: string;
  lastCommitSha: string;
  summary: string;
  runnerFinishedAt: number;
  minutesWatchersInterval: number;
  dependencies: DependencyWithVulnerabilities[];
  metrics: Metric[];
}


export interface Metric {
  id: number;
  severity: string;
  description: string;
  timestamp: number;
  repositoryId: number;
  watcherType: string;
}


export interface DependencyWithVulnerabilities {
  id: number;
  name: string;
  version: string;
  programmingLanguage: string;
  vulnerabilities: Vulnerability[];
}

export interface Vulnerability {
  id: number;
  cveId: string;
  cvePublishedAt: number;
  cveModifiedAt: number;
  cvss: number;
  cvssVector: string;
  cvePatch: string;
  dependencyId: number;
  riskIndex: number;
}
